plugin/src/resolve-gatsby-image-data.ts

```diff
@@ -50,13 +73,35 @@ export async function resolveGatsbyImageData(
     format: ext as ImageFormat,
   };
 
+  
+
+  if (remainingOptions && remainingOptions.placeholder === "BLURRED") {
+    // This function returns the URL for a 20px-wide image, to use as a blurred placeholder
+    const lowResImageURL = getLowResolutionImageURL({
+      ...remainingOptions,
+      formats,
+      layout,
+      sourceMetadata,
+      pluginName: `gatsby-source-shopify-experimental`,
+      filename: image.originalSrc,
+      generateImageSource,
+    })
+    const imageBase64 = await getImageBase64({
+      imageAddress: lowResImageURL,
+    })
+    
+    // This would be your own function to download and generate a low-resolution placeholder
+    remainingOptions.placeholderURL =  await getBase64DataURI({
+      imageBase64,
+    })
+  }
   return generateImageData({
-    ...options,
+    ...remainingOptions,
     formats,
     layout,
     sourceMetadata,
     pluginName: `gatsby-source-shopify-experimental`,
-    filename: basename,
+    filename: image.originalSrc,
```

Comment:
@ascorbic the issue with `?undefined` in the image URLs has popped up elsewhere too

https://github.com/gatsbyjs/gatsby-source-shopify-experimental/issues/95

So I want to make sure that our solution here will cover what that person is reporting. Armando filled me in a little bit on what we're changing here but it's not quite clear to me where we should or shouldn't include a version number in these URLs.